hexrd/ui/hand_drawn_mask_dialog.py

```diff
@@ -42,12 +43,21 @@ def __init__(self, canvas, parent):
     def setup_connections(self):
         self.ui.accepted.connect(self.accepted)
         self.ui.rejected.connect(self.rejected)
-        self.bp_id = self.canvas.mpl_connect('button_press_event',
-                                             self.button_pressed)
-        self.enter_id = self.canvas.mpl_connect('axes_enter_event',
-                                                self.axes_entered)
-        self.exit_id = self.canvas.mpl_connect('axes_leave_event',
-                                               self.axes_exited)
+        for canvas in self.parent().image_tab_widget.active_canvases:
```

Comment:
I think that expecting the parent to be a particular object can cause trouble if we move the dialogs around. It might be better to just pass the active canvases to this class' `__init__` method directly (similar to what we were doing before).

Reply:
The list of active canvases will change every time that we toggle the tabbed mode on/off or change the image mode. We can pass the list into the init function like we were before but it seems to me that we will still need a signal that emits the list of active canvases on image mode and tabbed mode change, unless we want to close the dialog and force the user to re-open it each time these are changed?

Reply:
So, this `setup_connections()` function is only called in the `__init__` method, so it looks like the code is currently looping over the active canvases at the time of `__init__`, and it is not currently updating the list of canvases if tabbed mode is toggled on/off or the image mode is changed.

Looking at the `ImageTabWidget` code, it looks like the first canvas is never deleted (it is used for all views and for the first tab in the tabbed raw view), but the other canvases (which are only used for non-first tabs in the tabbed raw view) are re-created when `clear()` is called (which happens several times).

I think in the main window function that creates this object, we can connect to any signals like `HexrdConfig().tab_images_changed` that result in the canvases changing, and then we can update the active canvases on this object from there.

Reply:
Sounds good, I think we're talking about the same approach. I am having some issues though so I've broken them out into their own comment for help/clarification.